helix-view/src/input.rs

```diff
@@ -4,8 +4,54 @@ use helix_core::unicode::{segmentation::UnicodeSegmentation, width::UnicodeWidth
 use serde::de::{self, Deserialize, Deserializer};
 use std::fmt;
 
-use crate::keyboard::{KeyCode, KeyModifiers};
+pub use crate::keyboard::{KeyCode, KeyModifiers};
+
+#[derive(Debug, PartialOrd, PartialEq, Eq, Clone, Copy, Hash)]
+pub enum Event {
+    Key(KeyEvent),
+    Mouse(MouseEvent),
+    Resize(u16, u16),
+    IdleTimeout,
```

Comment:
Is it good to only have a single idle timeout? Not sure if it would be useful to have two types, one long (~1s) and one short (~100ms).

Reply:
We definitely do need multiple idle timers, for example one for completion and other for debouncing inputs. I was hoping to tackle it in a separate PR.